### ISSUE TITLE
Activity and object use the same ID

### ISSUE THREAD

**Issue by puckipedia** [dariusk/express-activitypub#1]
When sending a message to a remote server, the ID used for the Create and the Note are equivalent. This is invalid, according to the spec, and will cause multiple servers to refuse it. (also it isn't dereferencable, which will cause issues if someone boosts the note, the servers where noone follows the bot will just not show the boost)

**Comment by dariusk** [comment]
Thanks, I've opened a PR that fixes the guids.

For the dereferenceable URIs, what should be served? Just the objects themselves? If someone goes to the Note id uri, will they just get the Note sub-object, and if someone goes to the Create id uri, they get served the Create object with the note sub-object?

Can I leave out dereferencing the Create and just dereference the Note?

**Comment by puckipedia** [comment]
for the Note's ID, you MUST (`RFC2119`) serve the Note, and for the Create, you MAY return the Create with the Note inside, or just the Create.

**Comment by dariusk** [comment]
@puckipedia I think the commit I just added addresses your concern. For example, if I send a Note with the `id` of `https://2914e630.ngrok.io/m/78c20ffafb39e7fedf91e0916d6ed893`, if you make a GET request to the URL with your browser you get the following JSON object (that URL won't work because I'm testing using a temporary domain):

```json
{
"id": "https://2914e630.ngrok.io/m/78c20ffafb39e7fedf91e0916d6ed893",
"type": "Note",
"published": "2018-09-25T03:06:47.111Z",
"attributedTo": "https://2914e630.ngrok.io/u/tester",
"content": "Just testing this, please ignore.",
"to": "https://www.w3.org/ns/activitystreams#Public"
}
```

This goes for both Create and Note objects. However, it doesn't address the boost issue, at least on Mastodon (if I boost this status from an account X on SERVER1, I can't see the boost from account Y on SERVER2 even though I follow X@SERVER1).

**Comment by dariusk** [comment]
Closed in #2.
